telegram/ext/filters.py

```diff
@@ -183,18 +183,57 @@ def __init__(self, name: Optional[str] = None, data_filter: bool = False):
         self._data_filter = data_filter
 
     def __and__(self, other: "BaseFilter") -> "BaseFilter":
+        """Defines `AND` bitwise operator for :class:`BaseFilter` object.
+        The combined filter accepts an update only if it is accepted by both filters.
+        For exampe, filters.PHOTO & filters.CAPTION will only accept messages that contain
+        both a photo and a caption
+
+        Returns:
+           :obj:`BaseFilter`
+        """
         return _MergedFilter(self, and_filter=other)
 
     def __or__(self, other: "BaseFilter") -> "BaseFilter":
+        """Defines `OR` bitwise operator for :class:`BaseFilter` object.
+        The combined filter accepts an update only if it is accepted by any of the filters.
+        For example, filters.PHOTO | filters.CAPTION will only accept messages that contain
+        photo or caption or both.
+
+        Returns:
+           :obj:`BaseFilter`
+        """
         return _MergedFilter(self, or_filter=other)
 
     def __xor__(self, other: "BaseFilter") -> "BaseFilter":
+        """Defines `XOR` bitwise operator for :class:`BaseFilter` object.
+        The combined filter accepts an update only if it is accepted by any of the filters and
+        not both of them. For example, filters.PHOTO ^ filters.CAPTION will only accept messages
+        that contain photo or caption, not both of them.
+
+        Returns:
+           :obj:`BaseFilter`
+        """
         return _XORFilter(self, other)
 
     def __invert__(self) -> "BaseFilter":
+        """Defines `NOT` bitwise operator for :class:`BaseFilter` object.
+        The combined filter accepts an update only if it is accepted by any of the filters.
+        For example, ~ filters.PHOTO will only accept messages that do not contain photo.
+
+        Returns:
+           :obj:`BaseFilter`
+        """
         return _InvertedFilter(self)
 
     def __repr__(self) -> str:
+        """Gives name for this filter.
+
+        .. seealso::
+               :meth:`name`
+
+        Returns:
+            :obj:`str`:
+        """
         return self.name
```

Comment:
`BaseFilter.__repr__` feels pretty useless. It should be updated in line with #3826 isn't it?

Reply:
Why is it useless? We have defined well readible names for all filters and some of these are in fact rather elaborate: https://github.com/python-telegram-bot/python-telegram-bot/blob/a0c81ec3d46fe1889ab3a9aca40b56f33784eabb/tests/ext/test_filters.py#L2037

Reply:
hm yeah I just forgot about that, I probably shouldn't review when I'm sleepy :P